DigitalLearningSolutions.Data/Services/TutorialContentService.cs

```diff
@@ -49,17 +49,39 @@ int tutorialId
                          Tutorials.Objectives,
                          Tutorials.VideoPath,
                          Tutorials.TutorialPath,
-                         Tutorials.SupportingMatsPath AS SupportingMaterialPath
+                         Tutorials.SupportingMatsPath AS SupportingMaterialPath,
+                         Sections.PLAssessPath AS PostLearningAssessmentPath,
+                         MIN(SubsequentTutorials.TutorialID) AS NextTutorialID,
+                         MIN(SubsequentSections.SectionID) AS NextSectionID
                     FROM Tutorials
                          INNER JOIN CustomisationTutorials
                          ON CustomisationTutorials.TutorialID = Tutorials.TutorialID
 
+                         INNER JOIN Sections
+                         ON Tutorials.SectionID = Sections.SectionID
+
                          INNER JOIN Customisations
                          ON CustomisationTutorials.CustomisationID = Customisations.CustomisationID
 
                          INNER JOIN Applications
                          ON Customisations.ApplicationId = Applications.ApplicationId
 
+                         LEFT JOIN CustomisationTutorials AS SubsequentCustomisationTutorials
+                         ON SubsequentCustomisationTutorials.CustomisationID = Customisations.CustomisationID
+                            AND SubsequentCustomisationTutorials.Status = 1
+
+                         LEFT JOIN Tutorials AS SubsequentTutorials
+                         ON SubsequentTutorials.SectionID = Sections.SectionID
+                            AND Tutorials.OrderByNumber < SubsequentTutorials.OrderByNumber
+                            AND SubsequentCustomisationTutorials.TutorialID = SubsequentTutorials.TutorialID
+
+                         LEFT JOIN Tutorials AS SubsequentSectionsTutorials
+                         ON  SubsequentCustomisationTutorials.TutorialID = SubsequentSectionsTutorials.TutorialID

```

Comment:
Why are we joining on Tutorials twice? 🙂

Reply:
The `SubsequentTutorials` is used to join on the other tutorials in this section (that are in `CustomisationTutorials`) but only if it has a greater `OrderBy`. This is done in a left join because that will give us a single row with `null` in this `NextTutorialId`, instead of giving us an empty table if we join first and then filter later. There might be more than one tutorial that matches though, so we aggregate them into a single row with the `MIN` function.

The second join on tutorials, `SubsequentSectionsTutorials` gives us all of the tutorials in the customisation (which we then use to find all the sections that are in the customisation, as opposed to the customisations in the application), which we then join `SubsequentSections` to give us rows for each section in the customisation.

At one point I did just have the simpler second join, but I think it would make finding the next tutorial in the section more difficult, because if the tutorial is the last in the section we want this field to be `null`, and not just `WHERE` on having a greater `OrderBy` and getting an empty table. There might be another way to do this but I don't know about it.

Does that make sense? I'm not sure how clear that all was 🙂

Reply:
>There might be more than one tutorial that matches though, so we aggregate them into a single row with the MIN function

So `SubsequentTutorials` gives all the subsequent tutorial in this section right? So if there's more than one tutorial after this one, how to we know that the MIN one is the next one? Shouldn't it be MIN based on the `OrderBy` instead of the `TutorialID`? 🙂 

The two joins on Tutorials make sense then, thanks 🙂

Reply:
Yep this is a problem, thanks for catching it! 🙂

Reply:
Fixed and regression tests added in 79d8087 🙂